### PR TITLE
To native

### PR DESCRIPTION
**Pull request by lorinbeer**
bunch of small changes

added to_native function with tests

removed py26 as supported environment

package bump to pylint

### CODE REVIEW

File: tests/yelp_bytes_test.py [hunk 1]
```diff
@@ -227,6 +227,20 @@ def test_to_bytes_is_like_str_encode(value):
     try:
         to_bytes_result = to_bytes(value, 'US-ASCII')
     except:
-        to_bytes_result = '(error)'
+        to_bytes_result = '(error)'  # pylint: disable=R0204
```
Comment (asottile) [inline tests/yelp_bytes_test.py]: use the long form of the disable `# pylint: disable=the-name-of-the-problem`

File: tests/yelp_bytes_test.py [hunk 2]
```diff
@@ -227,6 +227,20 @@ def test_to_bytes_is_like_str_encode(value):
...
+    UNICODE.latin1,
+    UNICODE.win1252,
+    UNICODE.bmp,
+    UNICODE.utf8,
+    b"this is a bytestring",
+    u"this is a unicode string",
+    "native string"
+))
+def test_to_native_with_unicode_objects(value):
+    assert to_native(value) == value.encode('UTF-8') if PY2 else value
```
Comment (asottile) [inline tests/yelp_bytes_test.py]: this *should* fail for b"this is a bytestring"

File: tests/yelp_bytes_test.py [hunk 3]
```diff
@@ -227,6 +227,20 @@ def test_to_bytes_is_like_str_encode(value):
...
+
+@pytest.mark.parametrize('value', (
+    UNICODE.ascii,
+    UNICODE.latin1,
+    UNICODE.win1252,
+    UNICODE.bmp,
+    UNICODE.utf8,
+    b"this is a bytestring",
+    u"this is a unicode string",
+    "native string"
```
Comment (asottile) [inline tests/yelp_bytes_test.py]: this file imports `unicode_literals` -- to be a native string you need `str("native string")`

File: tox.ini [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [tox]
 project = yelp_bytes
 # These should match the travis env list
-envlist = py26,py27,py33,py34,pypy,pypy3
+envlist = py27,py33,py34,pypy,pypy3
```
Comment (asottile) [inline tox.ini]: needs an update to .travis.yml (the comment above this line indicates such)

**Comment by asottile** [comment]
travis seems failing

**Comment by lorinbeer** [comment]
I've updated the branch to reflect feedback

I have not resolved the passing test which should fail, any ideas? The module behaves properly when run outside of tox venv

**Comment by asottile** [comment]
It boils down to this:

```python
assert to_native(value) == value.encode('UTF-8') if PY2 else value
```

I'll add parens to make it more obvious what's happening


```python
assert (to_native(value) == value.encode('UTF-8')) if PY2 else value
```

In python3 this simplifies to:

```python
assert value
```

needs more TDD

**Comment by bukzor** [comment]
shipit

File: yelp_bytes.py [hunk 5]
```diff
@@ -59,3 +61,13 @@ def to_utf8(obj, errors='strict'):
...
     """Decode utf8 bytes (str) to unicode text."""
     return from_bytes(obj, encoding='utf-8', errors=errors)
+
+
+def to_native(obj, errors='strict'):  # pragma: no cover
+    """ returns a native string regardless of py env """
+    if isinstance(obj, str):
+        return obj
+    elif PY2:
+        return obj.encode('UTF-8', errors)
```
Comment (bukzor) [inline yelp_bytes.py]: Lets use to_bytes.

File: yelp_bytes.py [hunk 6]
```diff
@@ -59,3 +61,13 @@ def to_utf8(obj, errors='strict'):
...
+    else:  # PY3
+        return obj.decode('UTF-8', errors)
```
Comment (bukzor) [inline yelp_bytes.py]: Let's use from_bytes.

These simple changes makes this function edge-case free.

File: tests/yelp_bytes_test.py [hunk 7]
```diff
@@ -230,3 +230,27 @@ def test_to_bytes_is_like_str_encode(value):
...
+
+@pytest.mark.parametrize('value', (
+    UNICODE.ascii,
+    UNICODE.latin1,
+    UNICODE.win1252,
+    UNICODE.bmp,
+    UNICODE.utf8,
+    u"this is a unicode string",
+    str("native string")
+))
```
Comment (bukzor) [inline tests/yelp_bytes_test.py]: You've only walked through cases of unicode with one ascii bytestring (for py2).

These need to be pumped up.

Throw in the "able" classes above, too.

**Comment by lorinbeer** [comment]
last addition, update travis.yml to match tox.ini
